tests/net/rpl/src/main.c

```diff
@@ -647,77 +598,41 @@ static bool test_dio_receive_dest(void)
 	prefix2 = net_if_ipv6_get_global_addr(&iface_def);
 
 	prefix = net_if_ipv6_get_global_addr(&iface);
-	if (!prefix) {
-		TC_ERROR("Will not send DAO as no global address was found.");
-		return false;
-	}
+	zassert_not_null(prefix,
+			"Will not send DAO as no global address was found.");
 
-	if (iface != iface_def) {
-		TC_ERROR("Network interface mismatch (%p vs %p)\n",
-			 iface, iface_def);
-		return false;
-	}
+	zassert_equal_ptr(iface, iface_def,
+			"Network interface mismatch");
 
-	if (prefix != prefix2) {
-		TC_ERROR("Network interface mismatch or not set (%p vs %p)\n",
-			 prefix, prefix2);
-		return false;
-	}
+	zassert_equal_ptr(prefix, prefix2,
+				"Network interface mismatch or not set");
 
 	msg_sending = NET_RPL_DODAG_INFO_OBJ;
 
 	ret = net_rpl_dio_send(iface, &instance, &peer_addr, &in6addr_my);
-	if (ret) {
-		TC_ERROR("%d: Cannot send DIO (%d)\n", __LINE__, ret);
-		return false;
-	}
+	zassert_false(ret, "cannot send DIO");
 
 	k_sem_take(&wait_data, WAIT_TIME);
 
 	if (data_failure) {
 		data_failure = false;
-		TC_ERROR("%d: Unexpected ICMPv6 code received\n", __LINE__);
-		return false;
+		zassert_true(0, "Unexpected ICMPv6 code received");
```

Comment:
as above

Reply:
fixed

Reply:
@andrewboie Can you please look at this PR once ?